### DISCUSSION TITLE
Can I Use Pymd3 (GPL-3.0 Licensed) in My MIT-Licensed Windows Application?

### DISCUSSION THREAD

**Discussion by mega6453** [doronz88/pymobiledevice3#1191] [Q&A] [answered]
Hi,
https://github.com/mega6453/AppiumWizard -- This is a windows app which I developed and using MIT license. 
If I want to use pymd3 in my project, will there be any problem as pymd3 uses GPL-3.0 license? 
**How I'll be using pymd3 :** 
I'll be creating windows exe of pymd3 and will be using it in my project. Won't be using your source code directly.

**Comment by doronz88** [comment, marked as answer]
As this project has started as a fork of the original [pymobiledevice](https://github.com/iOSForensics/pymobiledevice), the project's license will remain GPL.

However, after searching about the terms of service, from my understanding (mainly consulted ChatGPT), if you don't distribute the binary yourself, but instead make your software request it from another online source it is okay.

If your're willing, please submit a PR for building the artifacts - then you can reference the download links in your code.

ChatGPT: https://chatgpt.com/share/1af188fa-8ca6-48f3-bbbc-e672668efaff (focus on the first and last answers)

**Reply by mega6453** [reply]
First, Really thank you for taking time to do a research and answering to my question. I'll try to do as you've mentioned. Thanks again.

**Comment by nanoscopic** [comment]
If you are distributing your "MIT-licensed windows application", then no. You cannot mix gpl and mit code.

Eg: you can do that for yourself locally, but the moment you share such a thing you are violating license.

Usually people want MIT code so they can sell closed source applications using MIT code. You absolutely cannot legally add some GPL stuff into such a program.
